Automattic/woocommerce-payments
# Fix decimal error with shipping calculations with Multi-Currency.

Fixes #3158 

#### Changes proposed in this Pull Request

* Add new filter `fix_price_decimals_for_shipping_rates` for [`woocommerce_shipping_method_add_rate_args`](https://github.com/woocommerce/woocommerce/blob/5.8.0/includes/abstracts/abstract-wc-shipping-method.php#L266-L282)
  * This applies the decimals for the store currency to the shipping rate calculation instead of the current selected currency. If the currency selected had a different number of decimals, like USD vs JPY, it caused errors in the calculation.

#### Testing instructions

- Set the store currency to **USD**.
- Add **JPY** currency with a conversion rate of `109.98779`.
- Add two shipping **Flat Rates** of `60.26` and `81.70`.
- Choose **JPY** from the currency switcher.
- Go to **cart** and check the shipping converted price.
- Compare it with the single currency settings of **JPY**
- $60.26 US should convert to ¥6,700
- $81.70 US should convert to ¥9,000

-------------------


**Post merge**



- [] Added testing instructions to the [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) (or does not apply)
